### ISSUE TITLE
DOC: numpy.char.rjust cuts strings to size "width" instead of just padding like str.rjust does

### ISSUE THREAD

**Issue by Archern4r** [numpy/numpy#23895]
### Issue with current documentation:

Hello everyone, 

in the documentation https://numpy.org/doc/stable/reference/generated/numpy.char.rjust.html it is stated that numpy.char.rjust calls str.rjust element-wise.

Though, str.rjust returns the original string in case the length of the string is greater than the specified _width_ (https://docs.python.org/3/library/stdtypes.html#str.rjust) which is not the case for numpy.char.rjust, which cuts strings off if their length is greater than the given _width_. 

```python
import numpy as np
    
# numpy.char.rjust
arr = np.array(['Mississippi', 'Rome', 'Washington'])
result = np.char.rjust(arr, 7)
print(result)
# Output: ['Mississ' '   Rome' 'Washing']
```

### Idea or request for content:

This is hinted at by the explanation of the _width_-argument "The_ length of the resulting strings" but is different than the built-in function's behaviour and should be explained as such, in case this is even intentional.

To me, it seems more logical for the resulting strings not to be cut but instead only to be padded and to preserve the original's function's behaviour, but if that is not wanted for numpy, the documentation should reflect that. In that case, a short example or warning would be in order.

**Comment by rohan472000** [comment]
Yeah, agree on it, there should be an example for it for clear understanding.

**Comment by Archern4r** [comment]
@rohan472000 while this is mostly true, it does not address the issue that I pointed out: Unlike `str.rjust`, which [according to the documentation cited above](https://docs.python.org/3/library/stdtypes.html#str.rjust) just returns the original string, if it's longer than the specified _width_, numpy's version returns an `ndarray` with every string exactly of length _width_, even if that means cutting some of the string off. 

In my mind, this is not expected or wanted behaviour, but if it is, the documentation should mention it or warn about it. At the time of writing, the [docs](https://numpy.org/doc/stable/reference/generated/numpy.char.rjust.html) still state `str.rjust` is called element-wise, which suggests a different behaviour.

**Comment by Archern4r** [comment]
@rohan472000 did you write that original response and the updated one with ChatGPT? Because it has exactly the writing patterns of ChatGPT and it also does ChatGPT-like errors in the code and in its logic. 

In the example above, _apple_ is not truncated but instead it is padded. Also, your example with `str.rjust` is plain wrong, the output is not what python would actually return. Both examples do not display the problematic behaviour

While I appreciate the effort to help, I feel the issue will not be addressed by just rehearsing the functionalities of the two methods over and over.

Code to replicate the issue would be like:

        import numpy as np
        
        # numpy.char.rjust
        arr = np.array(['Mississippi', 'Rome', 'Washington'])
        result = np.char.rjust(arr, 7)
        print(result)
        # Output: ['Mississ' '   Rome' 'Washing']
        
        # str.rjust
        s = 'Washington'
        s_2 = 'Rome'
        print(s_2.rjust(7))
        print(s.rjust(7))
        # Output:    Rome
                  Washington

**Comment by rohan472000** [comment]
![Screenshot (210)](https://github.com/numpy/numpy/assets/96521078/2ea88e2c-ad85-4913-b0d1-ad94a51bfba7)


I recommend chatGPT only for documenting purpose, not for coding.

**Comment by Archern4r** [comment]
The `# Output: 'hello'`-line in your 2nd-answer example is not padded, although it is padded in your code screenshot - that's what I was referring to. ChatGPT itself claims to have written your 2nd answer. 

Let's please concentrate on the issue stated in the OP or any relevant information.

**Comment by seberg** [comment]
Well, rjust uses:
```
    a_arr = numpy.asarray(a)
    width_arr = numpy.asarray(width)
    size = int(numpy.max(width_arr.flat))
    if numpy.issubdtype(a_arr.dtype, numpy.bytes_):
        fillchar = asbytes(fillchar)
    return _vec_string(
        a_arr, type(a_arr.dtype)(size), 'rjust', (width_arr, fillchar))

```
TBH, I think it would be fair to set it to `max(size, a_arr.dtype.size)` (not valid code) with a release note.

**Comment by mattip** [comment]
Closing, we are deprecating `np.char`.
